resources/aws_ec2.go

```diff
@@ -158,30 +257,111 @@ func (obj *AwsEc2Res) Init() error {
 func (obj *AwsEc2Res) Watch() error {
 	send := false
 	var exit *error
-	if err := obj.Running(); err != nil {
-		return err
-	}
-	type chanStruct struct {
-		str string
-		err error
-	}
-	awsChan := make(chan *chanStruct)
-	closeChan := make(chan struct{})
-	ctx, cancel := context.WithCancel(context.TODO())
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
+	obj.awsChan = make(chan *chanStruct)
+	obj.closeChan = make(chan struct{})
+	obj.ctx, obj.cancel = context.WithCancel(context.TODO())
+	obj.wg = &sync.WaitGroup{}
+	obj.wg.Add(1)
 	go func() {
-		defer wg.Done()
+		defer obj.wg.Done()
 		select {
-		case <-closeChan:
-			cancel()
+		case <-obj.closeChan:
+			obj.cancel()
 		}
 	}()
-	wg.Add(1)
-	defer wg.Wait()
+	if obj.WatchEndpoint != "" {
```

Comment:
This part is good! However, do you think instead we should have these functions have the same signature as the main Watch and just do `return obj.longpollWatch()` for example? IOW is the rest of the Watch code sufficiently similar that it's worth sharing?